frontend/views/pages/Payments.vue

```diff
@@ -199,15 +201,24 @@ export default ({
           'lightning': L('Lightning'),
           'manual': L('Manual')
         }
+      },
+      historicalPayments: {
+        received: [],
+        sent: [],
+        todo: []
       }
     }
   },
   created () {
     this.setInitialActiveTab()
+    this.updatePayments()
   },
   watch: {
     needsIncome () {
       this.setInitialActiveTab()
+    },
+    ourPayments () {
+      this.updatePayments()
```

Comment:
Note: because `ourPayments` gets recomputed each time `reactiveDate.date` gets updated, this causes loading all of the payments history at that interval frequency, which in the previous diff patch I posted was once per second. That is too frequent to load all of the payments from disk!

So we should change this. Instead we should construct a custom watch statement that looks at the payments for the current period using: `getters.paymentHashesForPeriod(getters.periodStampGivenDate(getters.groupSettings.distributionDate))?.length`

Reply:
Nice catch. Thanks, @taoeffect

Reply:
@taoeffect, your solution breaks the cypress error. Even though we only watch the payments count, the way to get `todo` is using `currentPaymentPeriod`, I don't this could be best solution.
Please check my solution which get the historical payments only when `ourPayments` is changed.
I used `deepEqualJSONType` function from `giLodash`.